Common/UI/ViewGroup.cpp

```diff
@@ -411,22 +398,25 @@ NeighborResult ViewGroup::FindNeighbor(View *view, FocusDirection direction, Nei
 		}
 	}
 
-	// TODO: Do the cardinal directions right. Now we just map to
-	// prev/next.
+	if (direction == FOCUS_PREV || direction == FOCUS_NEXT) {
+		switch (direction) {
+		case FOCUS_PREV:
+			// If view not found, no neighbor to find.
+			if (num == -1)
+				return NeighborResult(0, 0.0f);
+			return NeighborResult(views_[(num + views_.size() - 1) % views_.size()], 0.0f);
+
+		case FOCUS_NEXT:
+			// If view not found, no neighbor to find.
+			if (num == -1)
+				return NeighborResult(0, 0.0f);
+			return NeighborResult(views_[(num + 1) % views_.size()], 0.0f);
+		default:
+			return NeighborResult(nullptr, 0.0f);
+		}
+	}
 
 	switch (direction) {
```

Comment:
One thing I can tell we differ on is, I really like to use the compiler warning that a switch is missing a case to warn me about places I've forgotten to handle a new value in an enum.  I try to avoid defaults or put asserts in them because I don't know what will go in there in the future and would rather have an error than silent breakage...

-[Unknown]

Reply:
I'm somehow in the habit of being super strict on this in Rust while not at all in C++ :P 

This change is kinda unnecessary, for a while it was a better cleanup until I discovered that we actually do use the "find the position of the view in the list" in other cases too, I tried to get it into this if. Should have reverted it all.